src/jit/optimizer.cpp

```diff
@@ -2851,6 +2852,23 @@ bool Compiler::optCanonicalizeLoop(unsigned char loopInd)
         JITDUMP("in optCanonicalizeLoop: redirect top predecessor BB%02u to BB%02u\n", topPredBlock->bbNum,
                 newT->bbNum);
         optRedirectBlock(topPredBlock, blockMap);
+
+        // When we have profile data then the 'newT' block will inherit topPredBlock profile weight
+        if (topPredBlock->hasProfileWeight())
+        {
+            // This corrects an issue when the topPredBlock has a profile based weight
+            //
+            JITDUMP(", hasProfileWeight() returns true");
+            assert(firstPred == true); // We currently don't expect to have multiple preds with profile weights
```

Comment:
I think it may be worth adding a jitstress run to make sure this assert does not fire. I am worried this may be another way to expose issue #13295.